File: app/src/debug/java/com/egoriku/catsrunning/DebugApplication.java
package com.egoriku.catsrunning;

import android.annotation.SuppressLint;
import android.support.multidex.MultiDexApplication;

import com.squareup.leakcanary.LeakCanary;

import timber.log.Timber;

@SuppressLint("Registered")
public class DebugApplication extends MultiDexApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());

        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }

        LeakCanary.install(this);
    }
}

File: app/src/debug/java/com/egoriku/catsrunning/DebugApplication.kt
package com.egoriku.catsrunning

import android.annotation.SuppressLint
import android.support.multidex.MultiDexApplication
import timber.log.Timber

@SuppressLint("Registered")
open class DebugApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        Timber.plant(Timber.DebugTree())
        DebugInitializer.register(this)
        LeakCanaryInitializer.register(this, false)
    }
}
